### ISSUE TITLE
[Bug] Icon reverts to blue for non-trusted sites

### ISSUE THREAD

**Issue by ChrisTheCat** [mrdokenny/Cookie-AutoDelete#1334]
### Acknowledgements

- [X] I acknowledge that I have read the above items

### Describe the bug

Icon turns blue for site with no matched domain expression and zero cookies. It seems like the correct behavior is for the icon to be red even if there aren't any cookies set by the site (yet).

### To Reproduce

Visit a site that doesn't set cookies. See icon briefly flash red then blue. Check to verify that there are no matched domain expressions.

### Expected Behavior

Icon remains red.

### Screenshots

<img width="800" alt="Screen Shot 2022-03-13 at 07 44 46" src="https://user-images.githubusercontent.com/8732435/158064969-e3ce7e5d-05a0-4b2c-b374-25c7b0a1c667.png">

### System Info - Operating System (OS)

macOS 12.2.1

### System Info - Browser Info

Firefox 91.7.0esr

### System Info - CookieAutoDelete Version

3.6.0

**Comment by kennethtran93** [comment]
This is currently expected due to a request from #66.  Your issue is also a duplicate of #1133.

Basically, to some users, when no cookies exist to clean up, there's not a point of having the icon go red (because we wouldn't need to clean it anyway unless other browsing data is enabled).  Perhaps a toggle to allow users to pick which icon color (red or blue) would appear if no cookies are set by the site.

By the way, if other browsing data is enabled, the cookie that we set when the site doesn't store cookies is excluded from the total cookie count.

**Comment by ChrisTheCat** [comment]
The problem for me is I visit a site where I want to allow cookies, but have not yet allowed them. The icon is blue, so I don't realize I haven't yet allowed cookies for the site. When I leave the site, the cookie that I wanted to save (my login, or whatever) isn't saved. But the icon was blue so it should have been saved, because that's what the icons are supposed to mean: blue = will save cookies and red = will not save cookies. Why in the world would you want blue to equal "might save cookies but might not" ?

**Comment by kennethtran93** [comment]
While I do understand where you are coming from, here's another way of looking at it. 

If the site hasn't set any cookies by the time you close the site, then there's a low chance of it setting a cookie after you leave the site. The icon will automatically turn red if the site you are currently on has set a cookie at any time while the page is still open.

Of course that's provided you keep/pin the icon on the toolbar and not hidden away.

**Comment by ChrisTheCat** [comment]
Right, but I looked at the icon when I first arrived at the site, it was
blue, so I logged in and yes, it turned red, but I wasn't going back to see
if it changed, I was assuming "it was blue, so it's a whitelisted site, so
after it sets a cookie that will be saved."

It would be great if there was an option in CAD for people to select the
behavior "icon color tells you if there is a rule or not, whether any
cookies have been set or not."

As it is, I'm setting the icon to "always grey" for now because I find it
to be misleading.

**Comment by BoffinBrain** [comment]
@ChrisTheCat your logic is sound. I have reported this 'bug' before myself over on #1133. I'm still baffled that #66 was ever approved, and I'm still looking forward to seeing it reverted, or an option added to make it behave in the sane way it used to.
